test/slam_manifold_preintegration_update_test.cc

```diff
@@ -0,0 +1,303 @@
+  const int steps_per_thread = sample_count / thread_count;
+  // NOTE(brad): I assume I can equally distribute samples among the threads
+  CHECK(sample_count % thread_count == 0);
+  std::vector<std::thread> threads;
+  std::array<Eigen::Matrix<double, 9, 9>, thread_count> covariance_sums;
+  covariance_sums.fill(M99::Zero());
+
+  Eigen::MatrixXd samples(9, sample_count);
+
+  for (int i = 0; i < thread_count; i++) {
+    threads.emplace_back([&samples, &covariance_sums, dt, i]() {
+      std::mt19937 gen(1914 + i);
+      // Each thread needs its own distribution as they are not thread safe
+      MultiVarNormalDist accel_dist(example::accel_cov / dt);
+      MultiVarNormalDist gyro_dist(example::gyro_cov / dt);
+
+      Eigen::MatrixXd accel_noise;
+      Eigen::MatrixXd gyro_noise;
+
+      for (int j = steps_per_thread * i; j < steps_per_thread * (i + 1); j++) {
+        sym::ImuPreintegrator<double> integrator(example::accel_bias, example::gyro_bias);
+
+        gyro_noise = gyro_dist.Sample(gen, measurements_per_sample);
+        accel_noise = accel_dist.Sample(gen, measurements_per_sample);
+
+        for (int k = 0; k < measurements_per_sample; k++) {
+          integrator.IntegrateMeasurement(example::true_accel + accel_noise.col(k),
+                                          example::true_gyro + gyro_noise.col(k),
+                                          example::accel_cov, example::gyro_cov, dt, 0.0);
+        }
+
+        samples.col(j).segment(0, 3) =
+            integrator.PreintegratedMeasurements().DR.ToTangent(sym::kDefaultEpsilond);
+        samples.col(j).segment(3, 3) = integrator.PreintegratedMeasurements().Dv;
+        samples.col(j).segment(6, 3) = integrator.PreintegratedMeasurements().Dp;
+
+        covariance_sums[i] += integrator.Covariance();
+      }
+    });
+  }
+  for (std::thread& t : threads) {
+    t.join();
+  }
+
+  const M99 calculated_covariance =
+      (1.0 / sample_count) *
+      std::accumulate(covariance_sums.begin(), covariance_sums.end(), M99::Zero().eval());
+
+  const Eigen::MatrixXd sampled_covariance = Covariance(samples);
+  const double sampled_covariance_max = sampled_covariance.array().abs().maxCoeff();
+
+  const Eigen::MatrixXd weighted_relative_error = sampled_covariance.binaryExpr(
+      calculated_covariance, [=](const double x, const double y) -> double {
+        return std::abs((x - y) / (std::abs(x) + 3e-2 * sampled_covariance_max));
+      });
+
+  CHECK(weighted_relative_error.maxCoeff() < 0.05);
+}
+
+TEST_CASE("Test preintegrated derivatives wrt IMU biases", "[slam]") {
+  using M33 = Eigen::Matrix<double, 3, 3>;
+  using M96 = Eigen::Matrix<double, 9, 6>;
+
+  const double dt = 1e-3;
+  const int iterations = 100;
+
+  sym::ImuPreintegrator<double> integrator(example::accel_bias, example::gyro_bias);
+
+  for (int k_ = 0; k_ < iterations; k_++) {
+    integrator.IntegrateMeasurement(example::true_accel, example::true_gyro, example::accel_cov,
```

Comment:
Shouldn't these also be shifted by the biases?  And ditto for the `IntegrateMeasurement` calls below? (I think by not adding in either place they do more-or-less cancel out, at least approximately).  I also wonder if it would be cleaner or clearer to just numerically differentiate the whole chain with `sym::NumericalDerivative`, which would handle loop to perturb along different dimensions?  This also seems ok though

Reply:
Huh. I didn't actually realize there was a `sym::NumericalDerivative` (I knew we calculated it for some tests, just not how precisely). I suppose the only aspect which is a pain is that in order to use `NumericalDerivative` we would have to break this out into 3 separate functions (or two, if we were to combine the rotational and another into a single function which returns a pose), one each for `DR`, `Dv`, and `Dp`. 

But yeah, as mentioned in the other comment, since a missing bias just translates to a different `true_accel` (which is chosen somewhat arbitrarily) (and since I forgot it equally everywhere), it doesn't affect whether or not the test passes. Still, I'll add it in (and thanks for catching it).